coder.ts

```diff
@@ -0,0 +1,218 @@
+    const files = await Promise.all(
+      entries.map(async entry => {
+        const res = path.resolve(dir, entry.name);
+        const relativePath = path.relative(directory, res);
+
+        if (ig.ignores(relativePath) || entry.name === ".gitignore") {
+          debug(`Ignoring file: ${relativePath}`);
+          return [];
+        }
+
+        if (entry.isDirectory()) {
+          return walkDir(res);
+        }
+
+        if (
+          entry.isFile() &&
+          (entry.name.endsWith(".md") || entry.name.endsWith(".mdx"))
+        ) {
+          if (await hasCodeComponentWithFileMeta(res)) {
+            debug(`Found file with code component: ${relativePath}`);
+            return res;
+          }
+          debug(
+            `Skipping file (no code component with file meta): ${relativePath}`,
+          );
+        }
+
+        return [];
+      }),
+    );
+    return files.flat();
+  };
+
+  return walkDir(directory);
+};
+
+const processContent = async (
+  content: string,
+  filePath: string,
+): Promise<string> => {
+  try {
+    const file = await unified()
+      .use(remarkParse)
+      .use(remarkFrontmatter)
+      // @ts-expect-error
+      .use(codeImport, {
+        preserveTrailingNewline: true,
+        removeRedundantIndentations: true,
+        rootDir: process.cwd(),
+      })
+      .use(remarkStringify, {
+        bullet: "-",
+        emphasis: "*",
+        fences: true,
+        listItemIndent: "one",
+        rule: "-",
+        ruleSpaces: false,
+        strong: "*",
+        tightDefinitions: true,
+      })
+      .process(content);
+    return String(file);
+  } catch (error) {
+    if ((error as NodeJS.ErrnoException).code === "ENOENT") {
+      throw new Error(
+        `File not found: ${(error as NodeJS.ErrnoException).path}`,
+      );
+    }
+    throw error;
+  }
+};
+
+const processFile = async (filePath: string): Promise<void> => {
+  try {
+    if (!(await hasCodeComponentWithFileMeta(filePath))) {
+      debug(`Skipping ${filePath}: No code component with file meta found.`);
+      return;
+    }
+
+    const originalContent = await fs.readFile(filePath, "utf8");
+    const processedContent = await processContent(originalContent, filePath);
+    if (originalContent !== processedContent) {
+      await fs.writeFile(filePath, processedContent);
+      console.log(`Updated: ${filePath}`);
+    } else {
+      debug(`No changes needed for: ${filePath}`);
+    }
+  } catch (error) {
+    console.error(`Error processing ${filePath}: ${(error as Error).message}`);
+  }
+};
+
+const processInChunks = async <T>(
+  items: T[],
+  processItem: (item: T) => Promise<void>,
+  chunkSize: number,
+): Promise<void> => {
+  for (let i = 0; i < items.length; i += chunkSize) {
+    const chunk = items.slice(i, i + chunkSize);
+    await Promise.all(chunk.map(processItem));
+  }
+};
+
+const watchFiles = async (directory: string): Promise<void> => {
```

Comment:
the file watcher does not seem to do anything for me?

Reply:
Watch flags update the code whenever any markdown file is changed.